hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMSecurityProtocolServer.java

```diff
@@ -151,6 +152,16 @@ public String getDataNodeCertificate(
     return getEncodedCertToString(certSignReq, NodeType.DATANODE);
   }
 
+  @Override
+  public String getReconCertificate(
+      NodeDetailsProto reconDetails,
+      String certSignReq) throws IOException {
+    LOGGER.info("Processing CSR for Recon {}, UUID: {}",
+        reconDetails.getHostName(), reconDetails.getUuid());
+    Objects.requireNonNull(reconDetails);
+    return getEncodedCertToString(certSignReq, NodeType.RECON);
```

Comment:
`NodeType` should be taken from the `NodeDetailsProto` parameter here.

Reply:
Sure, I will resolve this and update it with the comments above.